ssl/statem/statem_clnt.c

```diff
@@ -1136,6 +1136,21 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
                session_id_len);
     }
 
+    if (!ssl_verify_client_session_version(s)) {
```

Comment:
Well, we do get new session in all other error cases... I was just mimicking other places in this file ;)

So I can either set `s->hit = 0;` or remove `ssl_get_new_session` code. Which one would you prefer? (They seem to be the same to me).